dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -158,6 +160,20 @@ internal object CoreFeature {
         )
     }
 
+    fun drainExecutors() {
+        val tasks = ArrayList<Runnable>()
+        (persistenceExecutorService as? ThreadPoolExecutor)
+            ?.queue
+            ?.drainTo(tasks)
+        // we make sure we upload the currently locked files
+        uploadExecutorService
+            .queue
+            .drainTo(tasks)
+        tasks.forEach {
+            it.run()
```

Comment:
I'm still not feeling confident about letting the upload task add new runnables to the executor without shutting said executor. The fact that it did not happen in 10 small tests doesn't mean it'll never happen.